pennylane/control_flow/while_loop.py

```diff
@@ -154,6 +233,51 @@ def _(*args, args_slice, **__):
     return while_loop_prim
 
 
+def _add_abstract_shapes(f, shape_locations):
+    def new_f(*args, **kwargs):
+        results = f(*args, **kwargs)
+        new_shapes = [results[loc[0].arg_idx].shape[loc[0].shape_idx] for loc in shape_locations]
+        return new_shapes + results
+
+    return new_f
+
+
+def _has_dynamic_shape(val):
+    return any(not isinstance(s, int) for s in getattr(val, "shape", ()))
+
+
+def _get_dummy_arg(arg):
+    """If any axes are abstract, replace with an empty numpy array."""
+    if all(isinstance(s, int) for s in arg.shape):
+        return arg
+    shape = tuple(s if isinstance(s, int) else 2 for s in arg.shape)
+    return np.empty(shape=shape, dtype=arg.dtype)
+
+
+def _validate_no_resizing_returns(
+    jaxpr: "jax.core.Jaxpr", locations: list[list[AbstractShapeLocation]]
+) -> Optional[str]:
+    offset = len(locations)
+
+    for locations_list in locations:
+        l0 = locations_list[0]
```

Comment:
```suggestion
        l0 = locations_list[0]
```
Very minor, but this looks so much like the number 10 and kind of throws me - could we call this `loc0` instead?

Reply:
👍